### PR TITLE
BrainyQuote: Prevent triggering on 'stock quote'

### LINKED ISSUE #1952: BrainyQuote: Don't trigger on queries containing "stock quote"
In testing the Stock IA, I noticed that brainy quote triggers for any query that has "quote" and so it triggers on queries containing "stock quote" as well.

We should detect this in the `handle` function and bail early to prevent redundant API calls.

------
https://duck.co/ia/view/brainy_quote

### PR DESCRIPTION
**Pull request by kbhat95**
Seems rather inelegant, but it fixes #1952. Is there a better way to do this?

Also, I saw that this Spice was missing a tests file, so I added that in as well.

### CODE REVIEW

**Comment by moollaza** [comment]
>Also, I saw that this Spice was missing a tests file, so I added that in as well.

@kbhat95 I'm really surprised we were missing a test! Thanks a lot for making one.

File: lib/DDG/Spice/BrainyQuote.pm [hunk 1]
```diff
@@ -26,6 +26,10 @@ handle remainder => sub {
     if(/^\w\.\w/) {
         s/\./\. /g;
     }
+    
+    # Hacky fix to avoid triggering on 'stock quote'
+    # 'stock quote' is handled by the Stocks IA
+    s/stock//g;
```
Comment (moollaza) [inline lib/DDG/Spice/BrainyQuote.pm]: I guess I wasn't very clear, but this won't catch the case where the query is "apple stock quote" or similar.

So what we really want to do is check if the original query contains the word "stock quote" and if so, bail out with a `return;`

You can use the `$req` object which contains several useful properties, including `$req->query` which provides the original query and `$req->query_lc` which provides the lower-cased query.

So I suggest you run a regex against that for `m/stock quote/` and `return;` if it's true.

Comment (kbhat95) [inline lib/DDG/Spice/BrainyQuote.pm]: Oh, I see.

I'll get this done. :+1:

**Comment by kbhat95** [comment]
@moollaza funny story: it turns out that the 'apple stock quote' query also triggers the Recipes Spice. Should this be another issue?

File: lib/DDG/Spice/BrainyQuote.pm [hunk 2]
```diff
@@ -26,6 +26,11 @@ handle remainder => sub {
     if(/^\w\.\w/) {
         s/\./\. /g;
     }
+    
+    # Avoid triggering on 'stock' quotes; these are handled by Stocks IA
+    if ($req->query_lc =~ m/stock quote/) {
+       return;
+    }
```
Comment (iambibhas) [inline lib/DDG/Spice/BrainyQuote.pm]: Or you could just check if `$_` equals to `stock`.

Comment (kbhat95) [inline lib/DDG/Spice/BrainyQuote.pm]: Hmm, yes.

@moollaza @iambibhas , is there a significant gain in using that method, or does ```$req->query_lc``` do the trick? Disclaimer: new to Perl. :P

**Comment by jagtalon** [comment]
This looks good :+1:

**Pull request merged**